### ISSUE TITLE
Documentation for scss or another custom syntax usage

### ISSUE THREAD

**Issue by evilebottnawi** [stylelint/stylelint#543]
My code
```javascript
var rc = require('rc');
var reporter = require('postcss-reporter')
var stylelint = require('stylelint');
var processors = [
    stylelint({
        config: stylelintrc,
        syntax: 'scss'
    }),
    reporter(
       {
           clearMessages: true
       }
   )
];

return gulp.src(
    directory.src,
    {
        cwd: directory.cwd,
        base: directory.base,
        verbose: cli.verbose
    }
)
    .pipe($.postcss(processors));
```

I get error
```bash
Message:
    /path/to/scss/_variables.scss:11:15: Unknown word

$xs-strict: #{$xs-and-up} " and " #{$xs-and-down};
```
But if i include and set module manually
```javascript
var rc = require('rc');
var reporter = require('postcss-reporter')
var stylelint = require('stylelint');
var scss = require('postcss-scss');
var processors = [
    stylelint({
        config: stylelintrc,
        syntax: 'scss'
    }),
    reporter(
       {
           clearMessages: true
       }
   )
];

return gulp.src(
    directory.src,
    {
        cwd: directory.cwd,
        base: directory.base,
        verbose: cli.verbose
    }
)
    .pipe($.postcss(processors, { syntax: scss }));
```
All works fine.
Or it is normal set manually all plugins? Why, then, should be kept in dependency postcss-scss

**Comment by TrySound** [comment]
@evilebottnawi Syntax option is used only for standalone initialization. If you use stylelint as plugin this option doesn't have any effect.

**Comment by evilebottnawi** [comment]
@TrySound maybe add this information at docs?

**Comment by TrySound** [comment]
@evilebottnawi https://github.com/stylelint/stylelint/blob/master/docs/user-guide/node-api.md#syntax

**Comment by evilebottnawi** [comment]
@TrySound https://github.com/stylelint/stylelint/blob/master/docs/user-guide/postcss-plugin.md at this page, about syntax and example with scss syntax usage

**Comment by TrySound** [comment]
@evilebottnawi It's common for all postcss plugins
https://github.com/postcss/postcss#options

**Comment by evilebottnawi** [comment]
@TrySound Do not know about others, the documentation postcss I know well and its ecosystem. But I was a little discouraged - option syntax, depending on postcss-scss and the lack of documentation on this subject. I think I will not be the last who will ask the same question :/

**Comment by jeddy3** [comment]
> and the lack of documentation on this subject

I'm always eager to add documentation if it helps users out. Lets see what we can do here to make the relationship between `stylelint` and `postcss-scss`, and how it differs between the different APIs, a bit clearer.

> depending on postcss-scss

Am I right in thinking that we haven't provided a `syntax` option for the postcss plugin because 1. it's likely the users wants to apply the parser to all the plugins in the pipeline 2. it's the standard postcss way?

If so, lets, as per @evilebottnawi suggestion, expand out https://github.com/stylelint/stylelint/blob/master/docs/user-guide/postcss-plugin.md with a scss example (along with a link to https://github.com/postcss/postcss#options).

Sound good?

**Comment by TrySound** [comment]
>  it's likely the users wants to apply the parser to all the plugins in the pipeline
There can be only one parser/syntax in pipeline.

**Comment by jeddy3** [comment]
> There can be only one parser/syntax in pipeline.

Thanks for the clarification :) I'll knock up some docs.

**Comment by evilebottnawi** [comment]
@jeddy3 thanks for good support and adequate solution :+1:
